src/components/ToDoForm.vue

```diff
@@ -0,0 +1,45 @@
+<template>
+    <div class="todo-form">
+        <input v-model="currentItem"/>
+        <button @click="add()">Add item</button>
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+
+export default defineComponent({
+    props:{
+        addItem: {
+            type:Function, 
+            required:true
+        }
+    },
+    data() {
+        return {
+            currentItem: '',
+        }
+    },
+    methods:{
+        add(){
+            this.addItem({name:this.currentItem, completed:false});
+            this.currentItem = '';
+        }
```

Comment:
I don't think I've seen this before, passing a function as a prop. Is this something you've seen before?

I think normally you want to keep methods in the same component where you'll use them, unless you put them in a helper file.

Traditionally, your `add()` method should `$emit` an event to the parent, and in the parent you listen for the event.

Reply:
You are right, I implemented your solution and it works a lot better. 
This was something that I did when I worked on some react projects and it just became an habit.

Reply:
ah ok, interesting! I hope to work more with React in the future.